airsonic-main/src/test/java/org/airsonic/player/service/LegacyDatabaseStartupTestCase.java

```diff
@@ -2,53 +2,51 @@
 
 import org.airsonic.player.TestCaseUtils;
 import org.airsonic.player.util.HomeRule;
+import org.airsonic.player.util.MigrationConstantsRule;
 import org.apache.commons.io.FileUtils;
+import org.junit.BeforeClass;
 import org.junit.ClassRule;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.Description;
-import org.junit.runners.model.Statement;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.rules.SpringClassRule;
-import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import javax.sql.DataSource;
 
 import java.io.File;
 
-@ContextConfiguration(locations = {
-        "/applicationContext-service.xml",
-        "/applicationContext-cache.xml",
-        "/applicationContext-testdb.xml",
-        "/applicationContext-mockSonos.xml"})
-public class LegacyDatabaseStartupTestCase {
+import static org.assertj.core.api.Assertions.assertThat;
 
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ActiveProfiles({ "legacy" })
+public class LegacyDatabaseStartupTestCase {
     @ClassRule
-    public static final SpringClassRule classRule = new SpringClassRule() {
-        HomeRule airsonicRule = new HomeRule() {
-            @Override
-            protected void before() throws Throwable {
-                super.before();
-                String homeParent = TestCaseUtils.airsonicHomePathForTest();
-                System.setProperty("airsonic.home", TestCaseUtils.airsonicHomePathForTest());
-                TestCaseUtils.cleanAirsonicHomeForTest();
-                File dbDirectory = new File(homeParent, "/db");
-                FileUtils.forceMkdir(dbDirectory);
-                org.airsonic.player.util.FileUtils.copyResourcesRecursively(getClass().getResource("/db/pre-liquibase/db"), new File(homeParent));
```

Comment:
Minor bug here in that the DB copied over has the name "libresonic", but airsonic will search for and use "airsonic" in its default URL (fixed in the right)